### ISSUE TITLE
Trouble running FATES-SP mode with DEBUG on

### ISSUE THREAD

**Issue by ekluzek** [NGEET/fates#861]
A DEBUG on test case fails with FATES-SP mode as follows...

SMS_D_Ld30.f45_f45_mg37.I2000Clm50FatesCruRsGs.cheyenne_intel.clm-FatesColdDefReducedComplexSatPhen

This is in ctsm5.1.dev091 which uses: sci.1.56.0_api.23.0.0

```
85:forrtl: severe (408): fort: (3): Subscript #1 of the array T_VEG_PA has value 0 which is less than the lower bound of 1
85:
85:Image              PC                Routine            Line        Source             
85:cesm.exe           00000000045E4B0F  Unknown               Unknown  Unknown
85:cesm.exe           0000000001B2E0CD  fateshistoryinter        3620  FatesHistoryInterfaceMod.F90
85:cesm.exe           0000000000C2A6AA  clmfatesinterface        2353  clmfates_interfaceMod.F90
85:cesm.exe           0000000000ADAE4D  clm_driver_mp_clm        1101  clm_driver.F90
85:cesm.exe           00000000009F5032  lnd_comp_nuopc_mp         886  lnd_comp_nuopc.F90
85:libe
```

**Comment by ekluzek** [comment]
@rgknox and @glemieux I'm going to assume you two will work on this. If you'd like me to help let me know. When you have a fix in FATES, I'd like to try it out in the FATES-SP PR I have for turning soil BGC off.

**Comment by glemieux** [comment]
The issue here, I think, is that in when using nocomp + fbg (with or without `hlm_use_sp` being true), we set bare ground patches with a `patchno` equal zero in `set_patchno`:

https://github.com/NGEET/fates/blob/55d40f1836c08f36a4d1ce237075323d8dda81d9/biogeochem/EDPatchDynamicsMod.F90#L1326-L1329

Note that `nocomp_pft_label` above is set via `create_patch` prior to setting the `patchno` in `init_patches`:
https://github.com/NGEET/fates/blob/55d40f1836c08f36a4d1ce237075323d8dda81d9/main/EDInitMod.F90#L541-L554

Until recently, we haven't used `patchno` as an index in FatesHistoryInterfaceMod, and we haven't been running fates sp mode tests in debug mode, so we've missed it when [we added it](https://github.com/NGEET/fates/commit/faa6313a4bbb2431627ffaa03c8bdb642a4550e3).  

I think we can simply add a `nocomp_pft_label .ne. 0` check at the start of the patch loop in `update_history_hifrq` to avoid that segment of the output since it looks like there wouldn't be any relevant variables to output for bare ground data even in sp mode?  Is that correct  @ckoven @rosiealice?

Alternatively we could change the allocation statement for `t_veg_pa` to be from 0 to `maxPatchesPerSite` when we are using :
https://github.com/NGEET/fates/blob/55d40f1836c08f36a4d1ce237075323d8dda81d9/main/FatesInterfaceMod.F90#L506

**Comment by glemieux** [comment]
I tried out the [above fix](https://github.com/NGEET/fates/compare/master...glemieux:spmode-debug-fix) using https://github.com/ESCOMP/CTSM/pull/1747 and confirmed that a simple 1 year `1x1_brazil` test will build and run to completion in debug mode.

**Comment by ekluzek** [comment]
@glemieux can you try a global case for this with debug on and also a threaded global case?

**Comment by glemieux** [comment]
I ran a `f45` case using the FatesSp compset for 1 year using the latest ctsm tag and my `spmode_debug_fix` branch.  This ran successfully.  Location is here:
```
/glade/u/home/glemieux/scratch/ctsm-cases/issue861-gridded-debugcheck.fates-sci.1.56.0_api.23.0.0-ctsm5.1.dev096-Cd1da51e28-F90a475c7.intel
```

Threaded test forthcoming.

**Comment by glemieux** [comment]
I ran a couple of threading test as I my original test was taking a really long time to run and I wanted to see what might be causing the issue.  I varied the run type, duration, and thread count and in the end it appears that the `P32x2` setup is the cause of the issue.  All other variations on the `SMS_Ld3_P36x2.f45_f45_mg37.I2000Clm51FatesSpCruRsGs.cheyenne_intel.clm-FatesColdDefReducedComplexSatPhen` test case ran to completion in a reasonable amount of time (around a minute or so).  As of this writing the `P32x2` version is going on 26 mintues.

Passing test cases:
```
/glade/u/home/glemieux/scratch/ctsm-tests/tests_issue861-threaded
/glade/u/home/glemieux/scratch/ctsm-tests/tests_issue861-threaded-Ld30
/glade/u/home/glemieux/scratch/ctsm-tests/tests_issue861-threaded-ers
```
